desk/app/channels.hoon

```diff
@@ -386,6 +405,22 @@
   |=  [=mark =vase]
   ^+  cor
   ?+    mark  ~|(bad-poke+mark !!)
+      %noun
+    ?+  q.vase  !!
+        [%channel-wake @ @]
+      =+  ;;([=kind:c name=term] +.q.vase)
+      =/  =nest:c  [kind src.bowl name]
+      ?.  (~(has by v-channels) nest)  cor
+      ca-abet:(ca-safe-sub:(ca-abed:ca-core nest) |)
```

Comment:
should we maybe do a leave first here? or do we not need to?

Reply:
Not sure why we'd need to. If the subscription is already there, it's fine, it can just continue existing. We just want to (re-)establish it if it's not there, which is what `+ca-safe-sub` does.

Reply:
what if we think it's there, but they do not? will gall reject a duplicate sub going out like it does coming in?